File: implementations/php/MySDKless.php
<?php
require_once './sdkless/SDKless.php';

final class MySDKless extends SDKless\SDKless {

	const MERGE_OPEN = '*|'; // overwrite default if desired
	const MERGE_CLOSE = '|*';

	// local_vars can be used to provide api specific functionality that could not be standardized in SDKless
	public function go($endpoint_name, $endpoint_vars = array(), $local_vars = array()) {
		if (!is_array($local_vars))
			$local_vars = array();

		switch ($this->api_name) {
			case 'CampaignMonitor':
				switch ($endpoint_name) {
					case 'pull_list_contacts':
						// Campaign Monitor uses full name only; split into first/last
						$contacts = parent::go($endpoint_name, $endpoint_vars, $local_vars);
						
						foreach ($contacts as &$contact) {
							$name_parts = explode(' ', $contact['full_name']);
							$last_name = array_pop($name_parts);
							$first_name = implode(' ', $name_parts);
							$contact['first_name'] = $first_name;
							$contact['last_name'] = $last_name;
							unset($contact['full_name']);
						}

						return $contacts;
						break;
					case 'push_list_contacts':
						// Campaign Monitor uses full name only; concatenate first/last
						foreach ($endpoint_vars['array_set']['parameters']['contacts'] as &$value) {
							$value['full_name'] = $value['first_name'] . ' ' . $value['last_name'];
							unset($value['first_name']);
							unset($value['last_name']);
						}

						return parent::go($endpoint_name, $endpoint_vars, $local_vars);
						break;
					default:
						return parent::go($endpoint_name, $endpoint_vars, $local_vars);
				}

				break;
		}

		// default handler
		return parent::go($endpoint_name, $endpoint_vars, $local_vars);
	}

	private function remove_id_prefix(&$data, $prefix) {
		if (is_scalar($data))
			$data = str_replace($prefix, '', $data);
		
		if (is_array($data)) {
			foreach ($data as &$record) {
				if (is_array($record) && !empty($record['id']))
					$record['id'] = str_replace($prefix, '', $record['id']);

				if (is_object($record) && !empty($record->id))
					$record->id = str_replace($prefix, '', $record->id);
			}
		}
	}

}

File: implementations/django/my_project/my_app/MySDKless.py
from sdkless.SDKless import SDKless

class MySDKless(SDKless):

	MERGE_OPEN = '*|' # overwrite default if desired
	MERGE_CLOSE = '|*'

	def __init__(self, api_name = None, global_vars = {}):
		super(MySDKless, self).__init__(api_name, global_vars)

	# local_vars can be used to provide api specific functionality that could not be standardized in SDKless
	def go(self, endpoint_name, endpoint_vars = {}, local_vars = {}):
		if not isinstance(local_vars, dict):
			local_vars = {}

		if self.api_name == 'CampaignMonitor':
			if endpoint_name == 'pull_list_contacts':
				# Campaign Monitor uses full name only; split into first/last
				contacts = super(MySDKless, self).go(endpoint_name, endpoint_vars, local_vars)
				new_contacts = []

				for contact in contacts:
					name_parts = contact['full_name'].split()
					last_name = name_parts.pop(-1)
					first_name = " ".join(name_parts)
					contact['first_name'] = first_name
					contact['last_name'] = last_name
					del contact['full_name']
					new_contacts.append(contact)

				return new_contacts
			elif endpoint_name == 'push_list_contacts':
				# Campaign Monitor uses full name only; concatenate first/last
				for value in endpoint_vars['array_set']['parameters']['contacts']:
					value['full_name'] = "{} {}".format(value['first_name'], value['last_name'])
					del value['first_name']
					del value['last_name']

		# default handler
		return super(MySDKless, self).go(endpoint_name, endpoint_vars, local_vars)

	def _remove_id_prefix(self, data, prefix):
		new_data = data

		if isinstance(data, str):
			new_data = data.replace(prefix, '')
		
		if isinstance(data, list):
			for record in data:
				if isinstance(record, dict) and record.get('id'):
					record['id'] = record['id'].replace(prefix, '')

				new_data.append(record)

		return new_data
